### ISSUE TITLE
Methods that share a name implemented via macro call are incorrectly marked as errors when called

### ISSUE THREAD

**Issue by zphixon** [intellij-rust/intellij-rust#5127]
## Environment

* **IntelliJ Rust plugin version:** 0.2.118.2171-193
* **Rust toolchain version:** stable-x86_64-pc-windows-msvc 1.42
* **IDE name and version:** CLion 2019.3 (193.6911.21)
* **Operating system:** Win10

## Problem description

intellij-rust gives a false positive for methods that share a name when they are implemented via a macro call. The following code is reported as creating a compiler error despite being correct Rust.

```rust
macro_rules! bug {
    ($name:ident => $($variant:ident($kind:ty),)*) => {
        #[derive(Debug, Clone)]
        pub enum $name {
            $($variant($kind),)*
        }

        impl $name {
            pub fn into<T>(self) -> T where $name: Into<T> {
                std::convert::Into::into(self)
            }
        }

        $(impl Into<$kind> for $name {
            fn into(self) -> $kind {
                match self {
                    $name::$variant(t) => t,
                    _ => panic!("could not cast {:?} to {}", self, stringify!($kind)),
                }
            }
        })*
    };
}

bug!(E => A(i32),);

fn main() {
    let e = E::A(3);

    // call to std::convert::Into::<i32>::into
    let _: i32 = e.clone().into();

    // call to E::into::<i32>, incorrectly reported as a compiler error
    // intellij-rust: "Wrong number of type parameters: expected 0, found 1"
    let _ = e.clone().into::<i32>();
}
```

([playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f47abaf51445df88bb0ec33a382b6549))

This does not occur when manually implementing `E::into::<T>` and `std::convert::Into::<T>::into`.

**Comment by Undin** [comment]
@zphixon Thanks for the report!
The main reason of false-positive error annotation is that the default macro expansion engine can't process `impl` blocks generated by macro calls. It leads to wrong name resolution and as a result to false-positive error annotations.
Try to turn on new experimental macro expansion engine (see #3628 for more details). It should help
